lib/workers/branch/index.ts

```diff
@@ -363,134 +354,9 @@ export async function processBranch(
-                  stderr: sanitize(error.message),
-                });
-              }
-            } else {
-              logger.warn(
-                {
-                  cmd,
-                  allowedPostUpgradeCommands,
-                },
-                'Post-upgrade task did not match any on allowed list'
-              );
-              config.artifactErrors.push({
-                lockFile: upgrade.packageFile,
-                stderr: sanitize(
-                  `Post-upgrade command '${cmd}' does not match allowed pattern${
-                    allowedPostUpgradeCommands.length === 1 ? '' : 's'
-                  } ${allowedPostUpgradeCommands
-                    .map((x) => `'${x}'`)
-                    .join(', ')}`
-                ),
-              });
-            }
-          }
-
-          const status = await getRepoStatus();
-
-          for (const relativePath of status.modified.concat(status.not_added)) {
-            for (const pattern of fileFilters) {
-              if (minimatch(relativePath, pattern)) {
-                logger.debug(
-                  { file: relativePath, pattern },
-                  'Post-upgrade file saved'
-                );
-                const existingContent = await readLocalFile(relativePath);
-                const existingUpdatedArtifacts = config.updatedArtifacts.find(
-                  (ua) => ua.name === relativePath
-                );
-                if (existingUpdatedArtifacts) {
-                  existingUpdatedArtifacts.contents = existingContent;
-                } else {
-                  config.updatedArtifacts.push({
-                    name: relativePath,
-                    contents: existingContent,
-                  });
-                }
-                // If the file is deleted by a previous post-update command, remove the deletion from updatedArtifacts
-                config.updatedArtifacts = config.updatedArtifacts.filter(
-                  (ua) => ua.name !== '|delete|' || ua.contents !== relativePath
-                );
-              }
-            }
-          }
-
-          for (const relativePath of status.deleted || []) {
-            for (const pattern of fileFilters) {
-              if (minimatch(relativePath, pattern)) {
-                logger.debug(
-                  { file: relativePath, pattern },
-                  'Post-upgrade file removed'
-                );
-                config.updatedArtifacts.push({
-                  name: '|delete|',
-                  contents: relativePath,
-                });
-                // If the file is created or modified by a previous post-update command, remove the modification from updatedArtifacts
-                config.updatedArtifacts = config.updatedArtifacts.filter(
-                  (ua) => ua.name !== relativePath
-                );
-              }
-            }
-          }
-        }
-      }
+      config.updatedArtifacts = await executePostUpgradeCommands(config);
```

Comment:
Didn't this override any artifacts already generated by other artifacts functions?

Reply:
Inside `await executePostUpgradeCommands(config);` it copies the existing `config.updatedArtifacts` and then iterates over it. The only difference between old and new is that it no longer depends on in-memory editing of the updatedArtifacts object, instead it creates a copy at the start of the run and returns that.

This is also why I override the `updatedArtifacts` in the config passed to the 2nd call of `postUpgradeCommandExecutor`. i.e:

```
  const updatedArtifacts = await postUpgradeCommandExecutor(
    updateUpgradeCommands,
    config,
    getAdminConfig().allowPostUpgradeCommandTemplating
  );
  return postUpgradeCommandExecutor(
    branchUpgradeCommands,
    {
      ...config,
      updatedArtifacts,
    },
    false
  );
```

Reply:
On closer inspection of the code after replying to you I noticed that I'm not returning artifactErrors, so went and fixed that now it's:

```
return postUpgradeCommandsExecutor(
    branchUpgradeCommands,
    {
      ...config,
      updatedArtifacts,
      artifactErrors,
    },
    false
  );
```